File: algorithms/1_easy/T/00495_teemo_attacking/solution.c
int findPoisonedDuration(const int *restrict time_series, const int len, const int duration) {
    if (len <= 0 || time_series == NULL) { return 0; }

    int curr = 0;
    int res = 0;
    for (int i = 0; i < len; i += 1) {
        const int t = time_series[i];
        curr = MAX(curr, t);
        res += t + duration - curr;
        curr = t + duration;
    }

    return res;
}

File: algorithms/1_easy/T/00495_teemo_attacking/solution.rs
impl Solution {
    pub fn find_poisoned_duration(time_series: Vec<i32>, duration: i32) -> i32 {
        let mut curr = 0_i32;
        let mut res = 0_i32;
        for t in time_series {
            curr = curr.max(t);
            res += t + duration - curr;
            curr = t + duration;
        }

        res
    }
}
